src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -416,17 +414,34 @@ def symlink_or_copy(
 
     # Verify that the mapping isn't accidentally trying to create a file in the project source through symlinks.
     # We need to ensure we're resolving symlinks for this check to be effective.
+    # We are unlikely to hit this if calling the function through bundle map, keeping it here for other future use cases outside bundle.
     resolved_dst = dst.resolve()
     resolved_deploy_root = deploy_root.resolve()
-    if resolved_deploy_root not in resolved_dst.parents:
+    dst_is_deploy_root = resolved_deploy_root == resolved_dst
+    if (not dst_is_deploy_root) and (resolved_deploy_root not in resolved_dst.parents):
         raise NotInDeployRootError(dest_path=dst, deploy_root=deploy_root, src_path=src)
 
-    if overwrite:
+    absolute_src = resolve_without_follow(src)
+    if absolute_src.is_file():
         delete(dst)
-    try:
-        os.symlink(src, dst)
-    except OSError:
-        ssrc.copy(dst)
+        try:
+            os.symlink(absolute_src, dst)
+        except OSError:
+            ssrc.copy(dst)
+    else:
+        # 1. Create an empty shell directory in the deploy root
```

Comment:
nit: there is no step 2
other nit: it's not an empty shell, we just copy the directory, contents and all. It's just that the contents is symlinks. But that's still not empty.